### ISSUE TITLE
check error logging on multi threading

### ISSUE THREAD

**Issue by visr** [Deltares/Wflow.jl#139]
There was an issue with (wrong) negative f inputs that understandably caused a DomainError in the `log` function here:
https://github.com/Deltares/Wflow.jl/blob/af04ae4b8c9d2b65624a466a86f3b834b404be37/src/horizontal_process.jl#L85

However when running this code multithreaded, this error was not thrown. We should check why that is. #138 will help here as well.

**Comment by laurenebouaziz** [comment]
In hydromt, we recently fixed the calculation of the f parameter, which should now not produce negative values anymore (constrain M function was previously not applied to f).

**Comment by visr** [comment]
Yes, thanks, Willem showed me that you fixed it. Still created this issue to understand the interaction with error handling and multithreading a bit better.
